src/test/java/roomescape/application/ReservationServiceTest.java

```diff
@@ -0,0 +1,175 @@
+        );
+    }
+
+    @DisplayName("존재하지 않는 timeId로 예약 생성 시 예외 발생")
+    @Test
+    void error_when_timeId_notExist() {
+        // given
+        LocalDate date = LocalDate.now().plusDays(1);
+        String name = "멍구";
+        Long timeId = 999L;
+
+        ReservationRequest reservationRequest = new ReservationRequest(1L, date, name, timeId);
+
+        // when & then
+        assertThatThrownBy(() -> reservationService.registerReservation(reservationRequest))
+                .isInstanceOf(NotFoundException.class);
+    }
+
+    @DisplayName("중복된 일시에 예약 생성 시 예외 발생")
+    @Test
+    void error_when_dateTime_Duplicate() {
+        // given
+        LocalTime time = LocalTime.of(10, 0);
+        timeRepository.save(ReservationTime.withoutId(time));
+
+        LocalDate date = LocalDate.now().plusDays(1);
+        String name = "멍구";
+        Long timeId = 1L;
+
+        ReservationTime reservationTime = ReservationTime.of(1L, time);
+        Reservation reservation = Reservation.of(1L, name, THEME_1, date, reservationTime);
+        reservationRepository.save(reservation);
+
+        // when
+        ReservationRequest reservationRequest = new ReservationRequest(1L, date, "아이나", timeId);
+
+        // then
+        assertThatThrownBy(() -> reservationService.registerReservation(reservationRequest))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @DisplayName("모든 예약을 조회할 수 있다.")
+    @Test
+    void getAllReservations() {
+        // given
+        ReservationTime time1 = ReservationTime.of(1L, LocalTime.of(10, 0));
+        ReservationTime time2 = ReservationTime.of(2L, LocalTime.of(11, 0));
+
+        reservationRepository.save(Reservation.of(1L, "브라운", THEME_1, LocalDate.of(2024, 4, 1), time1));
+        reservationRepository.save(Reservation.of(2L, "솔라", THEME_1, LocalDate.of(2024, 4, 1), time2));
+        reservationRepository.save(Reservation.of(3L, "브리", THEME_1, LocalDate.of(2024, 4, 2), time1));
+
+        // when
+        List<ReservationDto> allReservations = reservationService.getAllReservations();
+
+        // then
+        assertAll(
+                () -> assertThat(allReservations).hasSize(3),
+                () -> assertThat(allReservations)
+                        .extracting(ReservationDto::name)
+                        .containsExactly("브라운", "솔라", "브리")
+        );
+    }
+
+    @DisplayName("예약을 id로 취소할 수 있다.")
+    @Test
+    void cancelReservation() {
+        // given
+        ReservationTime time = ReservationTime.of(1L, LocalTime.of(10, 0));
+        timeRepository.save(time);
+        reservationRepository.save(Reservation.of(1L, "브라운", THEME_1, LocalDate.of(2024, 4, 1), time));
+        assertThat(reservationRepository.findAll()).hasSize(1);
+
+        // when
+        reservationService.deleteReservation(1L);
+
+        // then
+        assertThat(reservationRepository.findAll()).hasSize(0);
```

Comment:
`isEmpty` 메소드도 있습니다.

```suggestion
        assertThat(reservationRepository.findAll()).isEmpty();
```

Reply:
`isEmpty()` 메서드가 검증하려는 대상이 비어있다는 의미를 더 직관적으로 드러낼 수 있다고 생각해 `isEmpty()` 메서드로 리팩터링 했습니다! 제안해주셔서 감사합니다 :)